binding.cc

```diff
@@ -370,18 +365,10 @@ std::string ResourceRecord::asAAAA() const {
 }
 
 std::string ResourceRecord::asCNAME() const {
-  if (record_->wType != DNS_TYPE_CNAME) {
-    throw std::runtime_error("Expected DNS CNAME record, received " +
-        std::to_string(record_->wType));
-  }
   return record_->Data.CNAME.pNameHost;
 }
 
 std::string ResourceRecord::asSRV() const {
-  if (record_->wType != DNS_TYPE_SRV) {
-    throw std::runtime_error("Expected DNS SRV record, received " +
-        std::to_string(record_->wType));
-  }
   const auto& srv = record_->Data.SRV;
```

Comment:
I am probably more confused than useful, but will this always be ok without the `record_->wType != DNS_TYPE_SRV` guard?

Reply:
I mean not that it wouldn't have been thrown also before I guess.

Reply:
Well, this is a private method that's only called when that condition is fulfilled now (because the switch happens over the actual record type, not the expected type like before). But you're right, it might be nice to keep assertions in place -- did that in e603c4ef1026 :)